### PR TITLE
Add multiple baseline and exclusion file support to APICompat

### PR DESCRIPTION
**Pull request by ericstj**
This permits more than one file for baselines and api lists.

Needed for https://github.com/dotnet/runtime/pull/33300

### CODE REVIEW

File: src/Microsoft.DotNet.ApiCompat/Program.cs [hunk 1]
```diff
@@ -199,19 +199,42 @@ bool RuleFilter(IDifferenceRuleMetadata ruleMetadata)
...
 
-        private static BaselineDifferenceFilter GetBaselineDifferenceFilter(string baselineFileName)
+        private static BaselineDifferenceFilter GetBaselineDifferenceFilter(string[] baselineFileNames)
         {
-            BaselineDifferenceFilter filter = null;
-            if (!string.IsNullOrEmpty(baselineFileName))
+            BaselineDifferenceFilter baselineDifferenceFilter = null;
+            
+            AddFiles(baselineFileNames, (file) => 
+                (baselineDifferenceFilter ??= new BaselineDifferenceFilter(new DifferenceFilter<IncompatibleDifference>())).AddBaselineFile(file));
```
Comment (safern) [inline src/Microsoft.DotNet.ApiCompat/Program.cs]: I might be missing something but I don't understand how new files will be added if baselineDifferenceFilter is not null.

Comment (safern) [reply src/Microsoft.DotNet.ApiCompat/Program.cs]: Duh, I miss a parenthesis when parsing it with my eyes. Never mind 😄

File: src/Microsoft.DotNet.ApiCompat/Program.cs [hunk 2]
```diff
@@ -199,19 +199,42 @@ bool RuleFilter(IDifferenceRuleMetadata ruleMetadata)
...
+
+
+                    addFile(file);
                 }
-                IDifferenceFilter incompatibleFilter = new DifferenceFilter<IncompatibleDifference>();
-                filter = new BaselineDifferenceFilter(incompatibleFilter, baselineFileName);
             }
-            return filter;
+
+            return;
```
Comment (safern) [inline src/Microsoft.DotNet.ApiCompat/Program.cs]: Is this return necessary? It is outside of the foreach.

File: src/Microsoft.DotNet.ApiCompat/Rules/Compat/AttributeDifference.cs [hunk 3]
```diff
@@ -1,4 +1,4 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (safern) [inline src/Microsoft.DotNet.ApiCompat/Rules/Compat/AttributeDifference.cs]: Did you intentionally change the BOM char?

Comment (ericstj) [reply src/Microsoft.DotNet.ApiCompat/Rules/Compat/AttributeDifference.cs]: VS removed it for some reason.  I can add it back.

**Pull request merged**